### PR TITLE
Fix pseudoknot structure handling in fullEnsemble

### PR DESCRIPTION
**Pull request by LunarFawn**
made function in fullEnsemble that uses the previously implemented pknots dot parens structure creation from the fold to address bug in subopt that did not display pknots correctly in the secondary structure

### CODE REVIEW

File: lib/CMakeLists.txt [hunk 1]
```diff
@@ -1,10 +1,10 @@
...
 
-add_subdirectory(Vienna1)
-add_subdirectory(Vienna2)
-add_subdirectory(LinearFold)
+#add_subdirectory(Vienna1)
+#add_subdirectory(Vienna2)
+#add_subdirectory(LinearFold)
 add_subdirectory(NUPACK)
-add_subdirectory(RNApuzzler)
+#add_subdirectory(RNApuzzler)
```
Comment (luxaritas) [inline lib/CMakeLists.txt:7]: Needs to be reverted

File: lib/NUPACK/emscripten/FullEnsemble.cpp [hunk 2]
```diff
@@ -181,17 +181,21 @@ FullAdvancedResult* FullEnsembleNoBindingSite(const std::string& seqString, int
...
+       // std::string singlestructure = "";             
+        //for ( int j = 0; j < suboptStructs.seqlength; j++ ) {
+        //    if( currentStruct.theStruct[j] > j ) {
+        //        singlestructure.push_back('(');
+        //    }
+        //    else if ( currentStruct.theStruct[j] == -1 ) {
+        //        singlestructure.push_back('.');
+        //    }
+         //   else singlestructure.push_back(')');
+        //}
```
Comment (luxaritas) [inline lib/NUPACK/emscripten/FullEnsemble.cpp:197]: Unused, remove?

File: lib/NUPACK/emscripten/FullEnsemble.cpp [hunk 3]
```diff
@@ -212,4 +216,92 @@ FullAdvancedResult* FullEnsembleNoBindingSite(const std::string& seqString, int
     return result;
 }
 
+//this is the current implementation of generating the dot paren structure from the fold as of 7/20/22
+//this seams to be lacking compaired to nupacks PrintDNAStructure called by the basics apps
+//when displaying the structure
+std::string getDotParens_OldMethod(bool pseudoknotted, const int seqlength, oneDnaStruct *currentStruct) {
```
Comment (luxaritas) [inline lib/NUPACK/emscripten/FullEnsemble.cpp:222]: * I see the separation you're trying to make, but please keep it camelcase all the way through
* Just call it getDotParens - we don't have an alternate implementation right now

File: src/eterna/folding/__tests__/Folder.test.ts [hunk 4]
```diff
@@ -31,14 +31,14 @@ test('NuPACK:suboptstructuresNoOligos', () => {
      .then((folder) => {
         if (folder === null) return;
 
-        let sequence = Sequence.fromSequenceString("auauauagaaaauauaua")
+        let sequence = Sequence.fromSequenceString("GGGAACGACUCGAGUAGAGUCGAAAAGAUAUAGAAAGCGCGACUAUAUCAAUAAACGCGCAAAAUAAAUAAAUAAAGAUCAGUUUCGACUGAUCAAAAGAAACAACAACAACAAC")
```
Comment (luxaritas) [inline src/eterna/folding/__tests__/Folder.test.ts:34]: It looks like you've modified existing tests, but we still want to test that use case! Please add a new test for the pkots behavior

**Comment by luxaritas** [comment]
Also please reset your changes to the package-lock.json - I'm pretty sure the current version on dev should be correct

**Comment by LunarFawn** [comment]
How do I reset my changes to package-lock.json? I was not aware I had
changed it..

**Pull request merged**